source/configuration_manual/authentication/ldap_conf_ext_settings.rst

```diff
@@ -0,0 +1,603 @@
+
+.. _ldap_conf_ext_setting-sieve_ldap_script_attr:
+
+``sieve_ldap_script_attr``
+--------------------------
+
+- Default: ``mailSieveRuleSource``
+- Values:  :ref:`string`
+
+Attribute containing the Sieve script.
+Used in the sieve script from ldap.
+
+
+Example:
+
+.. code-block:: none
+
+   sieve_ldap_script_attr = mailSieveRuleSource
+
+
+.. _ldap_conf_ext_setting-sieve_ldap_mod_attr:
+
+``sieve_ldap_mod_attr``
+-----------------------
+
+- Default: ``modifyTimestamp``
+- Values:  :ref:`string`
+
+Attribute used for modification tracking
+
+Example:
+
+.. code-block:: none
+
+   sieve_ldap_mod_attr = modifyTimestamp
+
+
+.. _ldap_conf_ext_setting-tls:
+
+``tls``
+-------
+
+- Default: ``no``
+- Values:  :ref:`boolean`
+
+Set to yes to use TLS to connect to the LDAP server.
+
+
+.. _ldap_conf_ext_setting-tls_ca_cert_file:
+
+``tls_ca_cert_file``
+--------------------
+
+- Default: <empty>
+- Values:  :ref:`string`
+
+Specify a value for TLS ``tls_ca_cert_file`` option.
+Currently supported only with OpenLDAP:
+
+
+.. _ldap_conf_ext_setting-tls_ca_cert_dir:
+
+``tls_ca_cert_dir``
+-------------------
+
+- Default: <empty>
+- Values:  :ref:`string`
+
+Specify a value for TLS ``tls_ca_cert_dir`` option.
+Currently supported only with OpenLDAP.
+
+
+.. _ldap_conf_ext_setting-tls_cipher_suite:
+
+``tls_cipher_suite``
+--------------------
+
+- Default: <empty>
+- Values:  :ref:`string`
+
+Specify a value for TLS ``tls_cipher_suite`` option.
+Currently supported only with OpenLDAP.
+
+
+.. _ldap_conf_ext_setting-tls_cert_file:
+
+``tls_cert_file``
+-----------------
+
+- Default: <empty>
+- Values:  :ref:`string`
+
+Specify a value for TLS ``tls_cert_file`` option.
+Currently supported only with OpenLDAP.
+
+
+.. _ldap_conf_ext_setting-tls_key_file:
+
+``tls_key_file``
+----------------
+
+- Default: <empty>
+- Values:  :ref:`string`
+
+Specify a value for TLS ``tls_key_file`` option.
+Currently supported only with OpenLDAP.
+
+
+.. _ldap_conf_ext_setting-tls_require_cert:
+
+``tls_require_cert``
+--------------------
+
+- Default: <empty>
+- Values: ``never, hard, demand, allow, try``
+
+Specify a value for TLS ``tls_require_cert`` option.
+Currently supported only with OpenLDAP.
+
+
+.. _ldap_conf_ext_setting-user_attrs:
+
+``user_attrs``
+--------------
+
+- Default: <empty>
+- Values:  :ref:`string`
+
+Specify user attributes to be retrived from LDAP (in userdb look up)
+See also :ref:`authentication-ldap_backend_configuration`
+User attributes are given in LDAP-name=dovecot-internal-name list.
+The internal names are:
+
+======== ========================
+name      Description
+======== ========================
+uid      System UID
+gid      System GID
+home     Home directory
+mail     Mail location
+======== ========================
+
+There are also other special fields which can be returned.
+
+See :ref:`authentication-user_extra_field`
+
+Example:
+
+.. code-block:: none
+
+   user_attrs = homeDirectory=home,uidNumber=uid,gidNumber=gid

```

Comment:
We should probably already recommend the "new way" of configuring these, which is more extensible.. Like:

```
user_attrs = \
  =home=%{ldap:homeDirectory}, \
  =uid=%{ldap:uidNumber}, \
  =gid=%{ldap:gidNumber}
```

Same with pass_attrs and iterate_attrs.

Reply:
fixed for user_attrs and pass_attrs. 
I did not change iterate_attrs, because sample configuration I could find in confluence was always like below;
 iterate_attrs = mailRoutingAddress=user